### PR TITLE
Fix user permissions baseline

### PR DESCRIPTION
**Pull request by mxtthias**
In preparation of #48, the following has been done:

* Remove public account registration
* Move registration to User model
* Update tests to use super user
* Add command to create user

### CODE REVIEW

File: tests/end2end/test_registering_user.py [hunk 1]
```diff
@@ -12,11 +12,17 @@
...
     # Goes to homepage
     res = testapp.get('/')
+    # Fills out login form in navbar
+    form = res.forms['loginForm']
+    form['username'] = superuser.email
+    form['password'] = 'myPrecious'
+    # Submits
+    form.submit().follow()
     # Clicks Create Account button
     res = res.click(_('Create account'))
```
Comment (mblomdahl) [inline tests/end2end/test_registering_user.py]: After login, a user can't go straight to "register user" form. First they must click to "Users" page, then they can find-and-click the "Create account" button (think this was one of the last things you removed).

File: xl_auth/commands.py [hunk 2]
```diff
@@ -95,6 +97,21 @@ def clean():
...
+@click.option('--password', default='password', help='Password for user (default: password)')
+@click.option('--active', default=True, is_flag=True, help='Activate account (default: True)')
+@click.option('--is-admin', default=False, is_flag=True, help='Create admin user (default: False)')
+@with_appcontext
+def create_user(email, full_name, password, active, is_admin):
+    """Create user account."""
+    full_name = full_name or email
+    user = User.create(email=email, full_name=full_name, password=password,
+                       active=active, is_admin=is_admin)
+    click.echo('Created account with login {0}'.format(user.email))
```
Comment (mblomdahl) [inline xl_auth/commands.py]: Suggestion: For people who doesn't read the help text defaults for password and just followed the README quickstart steps, how about printing out

    Created account with login <email>:<password>

?

File: tests/forms/test_user.py [hunk 3]
```diff
@@ -11,84 +11,87 @@
 
 
 # noinspection PyUnusedLocal
-def test_register_form_validate_without_full_name(db):
+def test_register_form_validate_without_full_name(superuser, db):
     """Attempt registering user with name shorter than 3 chars."""
-    form = RegisterForm(username='mr.librarian@kb.se', full_name='01',
+    form = RegisterForm(superuser, current_user=superuser, username='mr.librarian@kb.se',
```
Comment (mblomdahl) [inline tests/forms/test_user.py]: I think there is no `current_user` kwarg for *user.forms.RegisterForm*. WTForms is very forgiving... I think you could easily pass in `code_monkey=True` and it wouldn't say a word about it. Removing it?

**Comment by mxtthias** [comment]
@mblomdahl I've fixed your comments. Should I amend my previous changes or do you prefer a fix commit?

**Comment by mblomdahl** [comment]
@mxtthias Just push your new commits and let's see if it builds! There's no perfect commits in this repo to start with, so please lets not amend commits (or otherwise rewrite history) on origin! 😸

**Pull request merged**